src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2068 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.Editor.CSharp.CompleteStatement;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CompleteStatement;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CompleteStatement
+{
+    public class CSharpCompleteStatementCommandHandlerTests : AbstractCompleteStatementTests
+    {
+        private string CreateTestWithMethodCall(string code)
+        {
+            return
+@"class C
+    {
+        static void Main(string[] args)
+        {
+            int x = 1;
+            int y = 2;
+            int[] a = { 1,2 }
+            " + code + @"
+
+            int z = 4;
+        }
+    }
+
+    static class ClassC
+    {
+        internal static int MethodM(int a, int b)
+            => a * b;
+    }
+}";
+        }
+
+        #region ArgumentListOfMethodInvocation
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation1()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM($$x, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation3()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x,$$ y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
```

Comment:
these tests all follow the same pattern.  can we extract a helper function so that the just declaratively states the before/after, and says that it is testing semicolon typing?

Reply:
💭 We can, but I'm not sure we want to yet. We are hoping for Statement Completion to also handle cases where Automatic Brace Completion is disabled by inserting missing closing delimiters prior to inserting the final semicolon. However, there were a fairly large number of edge cases with bad behavior so we didn't include this functionality in the initial pull request.

Reply:
Improved the documentation for `VerifyTypingSemicolon`

Reply:
Yeah... i'd definitely like the tests to basically include all relevant parts of the command handling stack.  So that definitely includes (At a minimum):

1. formatting
2. brace completion
3. normal completion
4. "complete statement" (this feature).

Effectively, we should be verifying the behavior for the typing with all thse enabled for a whole host of scenarios.

Note: i'm not opposed to some targetted unit tests that just validate say the entire gamut of C# constructs we support that for.  However, we should take a set of interesting cases (invocation-completion, and maybe some other complex ones), and validate that it works properly across most of the permutations of the above.

Thanks!

Reply:
@CyrusNajmabadi Do we have any tests for permutations of the first three (the case where complete statement is disabled)?

Reply:
Will file as a follow-up issue.

Reply:
Here is the follow-up issue: https://github.com/dotnet/roslyn/issues/32126